core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1456,10 +1460,10 @@ class WorkspaceService(protected val ctx: RawlsRequestContext,
             } else Future.successful()
 
           _ <- workspaceAclManager.maybeShareWorkspaceNamespaceCompute(policyAdditions, workspaceName, ctx)
-          _ <- Future.traverse(policyRemovals.map(_._2)) { case email =>
-            dataSource.inTransaction { dataAccess =>
-              fastPassServiceConstructor(ctx, dataAccess).removeFastPassesForUserInWorkspace(workspace, email)
-            }
+
+          // Sync FastPass grants once ACLs are updated
+          _ <- Future.traverse(policyRemovals.map(_._2) ++ policyAdditions.map(_._2)) { case email =>
```

Comment:
Yes. `FastPassService` actually reaches out to Sam to get the user's permissions on the workspace. As a result, we don't need to pass how the users' access has changed, just that the users' access did change. `FastPassService` also checks Sam to see if the user is enabled, so that a disabled user doesn't get access to a workspace through being added.